src/workerd/io/tracer.h

```diff
@@ -79,6 +79,15 @@ class PipelineTracer: public kj::Refcounted {
 
   void addTailStreamWriter(kj::Own<tracing::TailStreamWriter>&& writer);
 
+  inline bool hasBuffered() {
+    return buffered;
+  }
+
+ protected:
```

Comment:
The protected thing here won't be needed once https://github.com/cloudflare/workerd/pull/5519 and the corresponding internal PR have landed.